tests/unit/test_ocolumn.py

```diff
@@ -29,7 +29,7 @@ def setUp(self):  # noqa: D102 (Missing docstring in public method)
 
     def test_creation(self):
         """Creating an OColumn instance works."""
-        OColumn([], 'attribute')
+        OColumn('attribute', [])
```

Comment:
I was disappointed that mypy didn't detect this bug, so I looked into strengthening the type checks in the tests.  That lead to a deep digression into the way I've done the typing for this module that hilighted several problems.  I think it's best to address those problems in a follow-up PR rather than mix it into this one.